app/features/TransactionSlice.ts

```diff
@@ -277,6 +279,44 @@ export const reloadTransactions = createAsyncThunk(
     }
 );
 
+/**
+ * Fetches a batch of the newest transactions of the given account,
+ * and saves them to the database, and updates the allDecrypted,
+ * if any shielded balance transactions were loaded.
+ */
+export async function fetchNewestTransactions(
+    dispatch: Dispatch,
+    account: Account
+) {
+    const newestTransactionInDatabase = await getTransaction(
+        account.maxTransactionId
+    );
+
+    if (
+        account.transactionFilter.toDate &&
+        secondsSinceUnixEpoch(
+            new Date(account.transactionFilter.toDate)
+        ).toString() < newestTransactionInDatabase.blockTime
```

Comment:
It is a little dirty to do the comparison with strings. It would be preferable that `newestTransactionInDatabase.blockTime` was considered as a date/number instead.